src/status_im/chat/models/input.cljs

```diff
@@ -0,0 +1,117 @@
+        commands  (mapv (fn [[_ command]]
+                          (vector command :any))
+                        commands)
+        responses (mapv (fn [{:keys [message-id type]}]
+                          (vector
+                            (get-in db [:chats chat-id :responses type])
+                            message-id))
+                        requests)]
+    (into commands responses)))
+
+(defn split-command-args [command-text]
+  (let [splitted (str/split command-text const/spacing-char)]
+    (-> (reduce (fn [[list command-started?] arg]
+                  (let [quotes-count       (count (filter #(= % const/arg-wrapping-char) arg))
+                        has-quote?         (and (= quotes-count 1)
+                                                (not= -1 (.indexOf arg const/arg-wrapping-char)))
+                        arg                (str/replace arg #"\"" "")
+                        new-list           (if command-started?
+                                             (let [index (dec (count list))]
+                                               (update list index str const/spacing-char arg))
+                                             (conj list arg))
+                        command-continues? (or (and command-started? (not has-quote?))
+                                               (and (not command-started?) has-quote?))]
+                    [new-list command-continues?]))
+                [[] false]
+                splitted)
+        (first))))
+
+(defn join-command-args [args]
+  (->> args
+       (map (fn [arg]
+              (if (= (.indexOf arg const/spacing-char) -1)
+                arg
+                (str const/arg-wrapping-char arg const/arg-wrapping-char))))
+       (str/join const/spacing-char)))
+
+(defn selected-chat-command [{:keys [current-chat-id] :as db} chat-id]
+  (let [chat-id          (or chat-id current-chat-id)
+        input-text       (get-in db [:chats chat-id :input-text])
+        input-metadata   (get-in db [:chats chat-id :input-metadata])
+        possible-actions (possible-chat-actions db chat-id)
+        command-args     (split-command-args input-text)
+        command-name     (first command-args)]
+    (when (.startsWith (or command-name "") const/command-char)
+      (when-let [command (-> (filter (fn [[{:keys [name]} message-id]]
+                                       (and (= name (subs command-name 1))
+                                            (= message-id (or (:to-message-id input-metadata)
+                                                              :any))))
+                                     possible-actions)
+                             (ffirst))]
+        {:command       command
+         :metadata      input-metadata
+         :args          (remove empty? (rest command-args))}))))
+
+(defn current-chat-argument-position
+  [command input-text]
+  (if command
+    (let [current (-> (:args command) (count))]
```

Comment:
better to use just `(count (:args command))`, or destruct `command` in parameters declaration and use `(count args)`, or at least change it to `(-> (:args command) count)`